test-resources/test/test-util/destination-service-mocks.ts

```diff
@@ -58,20 +59,168 @@ export function mockSubaccountDestinationsCall(
     .reply(responseCode, response);
 }
 
-export function mockSingleDestinationCall(
-  nockRef: nockFunction,
-  response: any,
-  responseCode: number,
-  destName: string,
-  headers: Record<string, any>,
-  options?: { uri?: string; badheaders?: string[] }
+type MockedDestinationConfiguration =
+  | (DestinationJson & {
+      owner: {
+        SubaccountId: string | null;
+        InstanceId: string | null;
+      };
+    })
+  | DestinationConfiguration;
+
+interface FindDestinationOptions {
+  response?: any;
+  responseCode?: number;
+  headers?: Record<string, any>;
+  destinationServiceUri?: string;
+  serviceToken?: string;
+  /**
+   * Mock a destination call with $skipTokenRetrieval. Defaults to `true`.
+   */
+  mockMetadataCall?: boolean;
+
+  /**
+   * Mock a destination call that includes the auth flow. If set to `true`, the metadata configuration will be reused.
+   * If a configuration is given, it overwrites the default.
+   */
+  mockAuthCall?:
+    | boolean
+    | {
+        response?: any;
+        responseCode?: number;
+        headers?: Record<string, any>;
+      };
```

Comment:
After looking at this and the implmentation below in more detail, I'd offer a different suggestion.

Firstly, as discussed, I wouldn't use the name "metadata" because it's not part of the Destination service's model. 
Secondly, I would actually consider simplifying what the function is supposed to do and put more of the complexity of the caller. Hear me out:

The production code has two functions, which are currently called `fetchDestination` and `fetchDestinationBytoken`. The mock function attempts to do both, depending on what parameters are required. This already could be considered as a mental mismatch. If I _know_ that there are two calls potentially being made in the backend, then I think it's fair two call `mockFindDestinationCall` two times in the test. This has the added benefit of communicating cleary to whoever reads the test afterwards, that there are two calls being made. In scenarios where I don't care specifically assert things on the individual nock scopes, this is a detail that could be missed more easily when the function mocks two calls.

In addition, you could simplify the implementation, because you don't need to check whether different http params are set for the mockAuthCall, you can just use the ones that are being passed via the existing parameters.

This would mean that instead of having two params `mockMetadataCall` and `mockAuthCall`, you'd only have one, which could be called `skipTokenRetrieval`. It could still default to true, and if it's set, you throw away the `authTokens` from the response and add the param to the uri. And if it's not set, you basically just mock the calls as defined by the params.

One more thing I find worth thinking about is aligning this API with the API of the production code. As said above, there are two separate functions in the production code, while there's only one mock function (even if you would refactor it the way I suggested). I think there's value in having an analogous API in both branches of the code. If I understood how the productive code works, I don't have to mentally map it to the test API. It's just the same. This would imply, though, that you either unify the production code, or split the test code into two mocking functions (at least in terms of its public API).

Finally, I'd align the names with the production code. I.e. either rename the prod code to `findDestination`, or rename the test code to `mockFetchDestinationCall`.

🖐️ 🎤

Reply:
The only thing I feel strongly about is the `metadata` naming thing. If you want to keep the function and API as is, I honestly don't care that much. But then I'd at least call the params `mockCallWithTokenRetrieval` and `mockCallWithoutTokenRetrieval` or something among those lines.